### PR TITLE
S3Client PUT and DELETE methods

### PR DESCRIPTION
**Pull request by SimonRichardson**
Adds PUT and DELETE methods to the s3caller client. The s3caller client is a wrapper around the s3client to ensure it works with the dependency engine.

The code adds simple PutObject and DeleteObject methods to the s3client. A new credentials type allows us to swap the credentials based on what the caller wants. This will be interesting to manage, as you'll likely want a s3client with different credentials.

For now, I've baked in the anonymous credentials and we can iterate on that.

Lastly, we should rip out the httprequest.Client from the API package. We shouldn't be forcing consumers of the API client to use some poorly maintained http.Client wrapper.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

Regression test to ensure we didn't break anything.

```sh
$ juju bootstrap lxd test --build-agent
$ juju deploy ubuntu
```

## Links

**Jira card:** JUJU-5291

### CODE REVIEW

File: internal/s3client/client.go [hunk 1]
```diff
@@ -13,38 +13,99 @@ import (
...
 )
 
 // Logger represents the logging methods called.
 type Logger interface {
-	Errorf(message string, args ...interface{})
-	Warningf(message string, args ...interface{})
-	Infof(message string, args ...interface{})
-	Debugf(message string, args ...interface{})
-	Tracef(message string, args ...interface{})
+	Errorf(message string, args ...any)
```
Comment (nvinuesa) [inline internal/s3client/client.go:23]: ✨

**Review by nvinuesa — COMMENTED** [review]
code looks good, but I don't see how the QA (only bootstrapping and deploying) tests this new functions

**Comment by SimonRichardson** [comment]
> Regression test to ensure we didn't break anything.

Regression test to ensure we didn't break anything.

**Review by nvinuesa — APPROVED** [review]
LGTM, indeed QA only needs to verify nothing broke. Thanks!

**Pull request merged**